schema_based_on_diagram.sql

```diff
@@ -0,0 +1,72 @@
+
+CREATE TABLE patients (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  UNIQUE (id),
+  name VARCHAR NOT NULL,
+  date_of_birth DATE NOT NULL
+);
+
+CREATE TABLE treatments (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  UNIQUE (id),
+  type VARCHAR NOT NULL,
+  name VARCHAR NOT NULL
+);
+
+CREATE TABLE medical_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP NOT NULL,
+  patient_id BIGINT REFERENCES patients(id),
+  treatment_id BIGINT REFERENCES treatments(id),
+  status VARCHAR NOT NULL,
+  CONSTRAINT fk_patient_id FOREIGN KEY(patient_id) REFERENCES patients(id),
+  CONSTRAINT fk_treatment_id FOREIGN KEY(treatment_id) REFERENCES treatments(id),
+  UNIQUE (patient_id),
+  UNIQUE (treatment_id)
+);
+-- Add Index to Foreign Keys
+CREATE INDEX fk_patient_id_medical_histories ON medical_histories(patient_id);
+CREATE INDEX fk_treatment_id_medical_histories ON medical_histories(treatment_id);
+
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  UNIQUE(id),
+  total_amount DECIMAL NOT NULL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP NOT NULL,
+  fk_medical_history_id BIGINT NOT NULL,
```

Comment:
- [ ] Kindly change the `fk_medical_history_id` to simply just `medical_history_id` so as to follow the design exactly 😄